src/player.h

```diff
@@ -1602,6 +1615,8 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         /** Stamp of skills. @ref learned_recipes are valid only with this set of skills. */
         mutable decltype( _skills ) valid_autolearn_skills;
 
+        /** smart pointer to targeting data stored for aiming the player's weapon across turns. */
+        std::shared_ptr<targeting_data> tdata;
```

Comment:
`shared_ptr` is for when you want multiple structures to keep the pointer, not just use it.
Since it is unlikely that you want multiple `player`s to have same `tdata`, `unique_ptr` is what you want here.